src/Build/BackEnd/Components/Logging/LoggingServiceLogMethods.cs

```diff
@@ -795,5 +797,11 @@ public void LogIncludeFile(BuildEventContext buildEventContext, string filePath)
         }
 
         #endregion
+
+        public void ProcessPropertyRead(PropertyReadInfo propertyReadInfo, AnalysisLoggingContext analysisContext)
```

Comment:
Question: I feel like processing property reads/writes in the logging service as a part of API is not a great option. Can't we send an event and then process them in the build check infrastructure?

Reply:
This is intentionall to simplify the switching between 'let's log the data and have BuildCheck consume them as build events' and 'let's consume the data with minimal performance overhead only for the build-in Checks, without exposing it via OM' - which was tried to be explained in the acompanying doc: https://github.com/dotnet/msbuild/pull/10237/files/dd73119b2b6a90587feab293150d48d0122d4ce0#diff-25077216bc2a35e0cdaafbba8e46b1f08ba6957b487e030139291a5f0c544ae5

Please have a look on the doc if it feeks it addresses the concern. If not - then let me know if the doc needs to be made more clear, or if you'd want to discuss the design choice

Reply:
I think your comment makes it a bit clearer to me than it is written in the docs. I am all for perf gains here. Although, I wonder, maybe instead of exposing the specific functions via the interface, let the logger service to expose a special helper class that will handle these. It might be the `BuildCheckManagerProvider` or a new helper class. I am concerned that for different small hooks that we will need to have in the code for all future build-in analyzers we will have separate small functions like those in multiple places. I want these hooks/functions to be concentrated in a build check related class, for them to be on a correct level of abstraction.

However, this is thinking about future. I am ok with current code given that this is an internal class, and we can refactor it at any time.

Reply:
Gotcha - makes sense! And that's how it was meant as well :-)

https://github.com/dotnet/msbuild/pull/10237/files#diff-12416749bc50c91e9746f77619c776d29683fd083b7964f3758d8671f7787a57R59

```
    internal interface ILoggingService : IBuildComponent, IBuildEngineDataRouter
    {
(...)

        /// <summary>
        /// Router of the build engine runtime execution information.
        /// </summary>
        IBuildEngineDataRouter BuildEngineDataRouter { get; }
```

This is what callers access on the ILoggingService in order to route the data.
It's just that currently the methods implementing this interface are members of the LoggingService - but in future they can be freely separated into separate class, without breaking any internal code contracts

Reply:
Ah, that makes sense!